c7n/resources/elasticsearch.py

```diff
@@ -133,6 +133,100 @@ def process(self, resources, event=None):
         return super().process(resources)
 
 
+@ElasticSearchDomain.filter_registry.register('cross-cluster')
+class ElasticSearchCrossClusterFilter(Filter):
+    """
+    Filter to return all elasticsearch domains with inbound cross-cluster with the given info
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: check-elasticsearch-cross-cluster
+            resource: aws.elasticsearch
+            filters:
+              - type: cross-cluster
+                inbound:
+                    key: SourceDomainInfo.OwnerId
+                    op: eq
+                    value: '123456789'
+                outbound:
+                    key: SourceDomainInfo.OwnerId
+                    op: eq
+                    value: '123456789'
+    """
+    schema = type_schema(type_name="cross-cluster",
+        inbound={
```

Comment:
Looks like the intent of this schema is to say "inbound and outbound can do whatever the value filter can do", which is great :+1: . We should be able to cover the same ground and automatically handle any future value filter updates by using something like this:

```
schema = type_schema(type_name="cross-cluster",
    inbound=type_schema(
        type_name='inbound',
        required=('key', 'value'),
        rinherit=ValueFilter.schema
    ),
    outbound=type_schema(
        type_name='outbound',
        required=('key', 'value'),
        rinherit=ValueFilter.schema
    ),
)
```

Reply:
Hi @ajkerrigan thanks you for the feedback! I have made the requested changes and retested the filter.

Reply:
Thanks for making those changes @cjhallman ! On a second look at this PR I noticed that there were Terraform bits (module and state files) that weren't being referenced anywhere. We can likely either remove those bits or work them into the tests. I added a commit showing an example of how we could work them in... feel free to keep that or ditch it. Appreciate you doing the Terraform work and sharing it as part of your testing :+1:

Reply:
Hi @ajkerrigan thanks for reviewing and yea I mostly just used terraform as a way to spin up some resources for testing but I appreciate you working them into the test cases. I think I will leave them in there. I have a question for you, how did you generate tf_resources.json file included in your commit?

Reply:
I also made one more commit with terraform changes to actually create the cross cluster connection

Reply:
I manually did what pytest-terraform does under the covers [here](https://github.com/cloud-custodian/pytest-terraform/blob/adeacc755277fa5ff0c59c0fe61784501c8432a9/pytest_terraform/tf.py#L245-L284) to parse & save terraform state, using the state file you committed.

Most tests don't include the Terraform bits, and elasticsearch clusters take such a long time to launch that we're unlikely to exercise them in live functional tests. But a template and state recording is a helpful reference for folks who might need to work with this filter in the future.